src/main/java/org/gridsuite/modification/server/modifications/AbstractScaling.java

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.modification.server.modifications;
+
+import com.powsybl.commons.reporter.Reporter;
+import com.powsybl.commons.reporter.TypedValue;
+import com.powsybl.iidm.modification.scalable.Scalable;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.network.store.iidm.impl.NetworkImpl;
+import org.gridsuite.modification.server.ModificationType;
+import org.gridsuite.modification.server.NetworkModificationException;
+import org.gridsuite.modification.server.dto.FilterEquipments;
+import org.gridsuite.modification.server.dto.FilterInfos;
+import org.gridsuite.modification.server.dto.IdentifiableAttributes;
+import org.gridsuite.modification.server.dto.ScalingInfos;
+import org.gridsuite.modification.server.dto.ScalingVariationInfos;
+import org.gridsuite.modification.server.service.FilterService;
+import org.gridsuite.modification.server.service.SpringContext;
+import org.springframework.util.CollectionUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+import static org.gridsuite.modification.server.modifications.ScalingUtils.createReport;
+
+/**
+ * @author bendaamerahm <ahmed.bendaamer at rte-france.com>
+ */
+public abstract class AbstractScaling extends AbstractModification {
+    protected final ScalingInfos scalingInfos;
+    private static final String ERROR_MESSAGE = "this variation mode is not supported";
+
+    protected AbstractScaling(ScalingInfos scalingInfos) {
+        this.scalingInfos = scalingInfos;
+    }
+
+    @Override
+    public void apply(Network network, Reporter subReporter) {
```

Comment:
I think we can simplfy the code to facilitate understanding

Reply:
ill see what i can do to simplify the code